### ISSUE TITLE
Add von Mises-Fisher loss

### ISSUE THREAD

**Issue by erip** [OpenNMT/OpenNMT-py#1209]
[This paper](https://arxiv.org/abs/1812.04616) claims to see a 2.5x training speedup using a von Mises-Fisher loss function on seq2seq problems. Luckily, their experimental testbed was using this repo. 😄 Their code can be found [here](https://github.com/Sachin19/seq2seq-con) which defines losses as functions instead of `nn.Module`s. It seems like this might be neat to add if it really is this much faster.

**Comment by vince62s** [comment]
This paper is misleading to me.
It does not really speedup, it just "seems" to converge faster.
it's bit like adafactor or adaptive loss, but in the end no revolution.
